### ISSUE TITLE
nexus-inference: multi-layer RNN convenience API

### ISSUE THREAD

**Issue by Abso1ut3Zer0** [Abso1ut3Zer0/nexus#350]
## Context

`TinyLstmF32` and `TinyGruF32` are single-layer, unidirectional, streaming RNNs. PyTorch's `nn.LSTM` and `nn.GRU` support `num_layers > 1` natively, stacking layers with optional dropout between them.

## Current offering

- Single-layer LSTM and GRU with fused gate matrices
- AVX2/AVX-512 vectorized sigmoid/tanh (Padé [7,6])
- Streaming `step` / `step_into` API
- PyTorch `nn.LSTM` / `nn.GRU` weight layout

## What's composable today

Multi-layer RNNs can be built by chaining instances:

```rust
let lstm1 = TinyLstmF32::new(input_size, hidden_size, &w1);
let lstm2 = TinyLstmF32::new(hidden_size, hidden_size, &w2);

// Per timestep:
lstm1.step_into(input, &mut h1, &mut c1, &mut mid);
lstm2.step_into(&mid, &mut h2, &mut c2, &mut output);
```

This works but requires the user to manage intermediate buffers and per-layer state manually.

## Proposed enhancement

A convenience wrapper that:
- Holds N single-layer RNNs internally
- Manages intermediate buffers and per-layer hidden/cell state
- Loads from a single PyTorch multi-layer state_dict (which uses `weight_ih_l0`, `weight_ih_l1`, etc.)
- Exposes the same `step` / `step_into` streaming API

## What we intentionally don't support

- **Bidirectional:** Requires future context, incompatible with streaming inference. Users needing bidirectional should process complete sequences offline.
- **Dropout between layers:** No-op at inference time. PyTorch applies dropout only during training (`model.train()`), never during `model.eval()`.
- **Projection (LSTM only):** `proj_size` parameter in PyTorch adds a linear projection after each layer. Niche optimization for reducing parameter count — can be added later if needed.

## Limitation for users

**Until this is implemented:** Users training multi-layer RNNs in PyTorch (`num_layers > 1`) must manually split the state_dict by layer index and create separate `TinyLstmF32` / `TinyGruF32` instances. Single-layer models (`num_layers=1`, the default) work as-is.

## Priority

Ergonomic improvement. The feature is fully composable today — this is about reducing boilerplate for multi-layer models.

**Comment by Abso1ut3Zer0** [comment]
Implemented in PR #356. StackedLstmF32 and StackedGruF32 with from_parts, from_safetensors (auto-detecting num_layers), full PyTorch parity tests, and benchmarks.
